File: iOSApp/Collegefeed/Collegefeed/College.m
//
//  College.m
//  Collegefeed
//
//  Created by Patrick Sheehan on 5/6/14.
//  Copyright (c) 2014 Appuccino. All rights reserved.
//

#import "College.h"

@implementation College

- (id)initWithCollegeID:(NSInteger)cID withName:(NSString*)cName
{
    self = [super init];
    if (self)
    {
        [self setCollegeID:cID];
        [self setName:cName];
        
        [self validateCollege];
        return self;
    }
    return nil;
}
- (id)initWithCollegeID:(NSInteger)cID withName:(NSString*)cName
                withLat:(NSInteger)lat withLon:(NSInteger)lon
{
    self = [super init];
    if (self)
    {
        [self setCollegeID:cID];
        [self setName:cName];
        [self setLat:lat];
        [self setLon:lon];
        
        [self validateCollege];
        return self;
    }
    return nil;
}

- (id)initWithCollegeID:(NSInteger)cID withName:(NSString*)cName withShortName:(NSString*)cShortName
{
    self = [super init];
    if (self)
    {
        [self setCollegeID:cID];
        [self setName:cName];
        [self setShortName:cShortName];
        
        [self validateCollege];
        return self;
    }
    return nil;
}
- (void)validateCollege
{   //TODO: ensure college is valid here

//    if (false)
//    {
//        [NSException raise:@"Invalid College" format:@"College \"%@\" invalid", self.name];
//    }
}

@end

File: AndroidApp/src/com/appuccino/collegefeed/objects/College.java
package com.appuccino.collegefeed.objects;

public class College 
{
	String name;
	String shortName;	//optional
	int id;
	double latitude;
	double longitude;
	
	public College(String name)
	{
		this.name = name;
	}
	
	public College(String name, int id, double lat, double lon)
	{
		this.name = name;
		this.id = id;
		this.latitude = lat;
		this.longitude = lon;
	}
	
	public String getName()
	{
		return name;
	}

	@Override
	public String toString() {
		return "College name=" + name + ", shortName=" + shortName + ", id="
				+ id + ", latitude=" + latitude + ", longitude=" + longitude;
	}
}
